src/Control/Controller.php

```diff
@@ -332,14 +326,10 @@ public function getURLParams()
     /**
      * Returns the HTTPResponse object that this controller is building up. Can be used to set the
      * status code and headers.
-     *
-     * @return HTTPResponse
      */
-    public function getResponse()
+    public function getResponse(): HTTPResponse
     {
-        if (!$this->response) {
-            $this->setResponse(new HTTPResponse());
-        }
+        $this->response ??= HTTPResponse::create();
```

Comment:
`setResponse()` should be used here

Reply:
Have updated to use `$this->setResponse(HTTPResponse::create())` in the constructor.